lang/types/value.go

```diff
@@ -65,17 +70,23 @@ func ValueOfGolang(i interface{}) (Value, error) {
 // a resource field. This is done with our "elvis" operator. When using this
 // function, if you pass in something with a nil value, then expect a panic or
 // an error if you're lucky.
-// FIXME: we should make sure we error instead of ever panic-ing on a nil value
-func ValueOf(v reflect.Value) (Value, error) {
-	value := v
+func ValueOf(value reflect.Value) (Value, error) {
+	// Gracefully handle invalid values instead of panic().
+	if !value.IsValid() {
```

Comment:
Are you sure we want IsValid ?

Reply:
Just checking the docs, yes. .Elem() can return the zero value and IsValid() returns false for the zero value. Will add a comment to the code

Reply:
Kept the IsValid check, but dropped the recursive call. Clarified with a comment above the check.